pattern-system.md

```diff
@@ -0,0 +1,273 @@
+
+They classify patterns so that they "can refer to families of related patterns."
+The authors also suspect that this classification will make "learning the
+catalog" faster. They Classify patterns according to two criteria: Purpose and
+Scope. They introduce the following classes of purpose in their book:
+
+- Creational, 
+- Structural and
+- Behavioral.
+
+Orthogonal to this, they distinguish two scope classes, related to object
+oriented SW development:
+
+- Class and
+- Object.
+
+The main ordering criterion in the books catalog is the Purpose and each
+pattern is "tagged" with the scope classification.
+
+### Pattern-Oriented Software Architecture (Fram Buschmann et. al.)
+
+Frank Buschmann also organizes the patterns in his (first) book using, what he
+calls "categories". Similarly to the GoF book he uses one category to organize
+the content of his book, in this case the level of abstraction the pattern
+applies on:
+
+- Architectural Patterns,
+- Design Patterns and
+- Idioms.
+
+The second criterion categorizes the problems solved by the patterns:
+
+- _From Mud to Structure_,
+- _Distributed Systems_,
+- _Interactive Systems_, 
+- _Adaptable Systems_,
+- _Structural Decomposition_,
+- ... 
+
+you get the point.
+
+In addition to this, he postulates a number of properties, a pattern system 
+must exhibit. According to Buschmann, a pattern system
+
+- should be simple and easy to learn,
+- should consist of only a few classification criteria,
+- should use a classification criterion that the reflects _natural properties_
+  of patterns (e. g. kinds of problems, not e. g. what pattern language a
+  patterns belongs to),
+- should provide a _roadmap_ that leads users to a set of potentially
+  applicable patterns and
+- it should be open to the integration of new patterns.
+
+### Conclusion
+
+I have used both books extensively and found their respective pattern systems
+very helpful in finding patterns for specific problems and for finding related
+patterns. I also think that the criterions for a useful pattern system
+postulated by Frank Buschmanns are sensible and might serve as a role model for
+our InnerSource pattern system.
+
+### A pattern language for creating pattern languages (Takashi Iba)
+
+Takashi Iba has published an article in the ACM Digital Library from PLoP 2016:
+[A pattern language for creating pattern languages: 364 patterns for pattern
+mining, writing, and symbolizing](https://dl.acm.org/citation.cfm?id=3158175&CFID=831673585&CFTOKEN=74341142&qualifier=LU1011674)
+- for those without ACM DL access, there is [an earlier draft of the paper from
+PLoP 2016](http://www.hillside.net/plop/2016/papers/three/26.3.pdf).
+
+## Candiate Classifications
+
+This section shall serve to collect individual proposals for systems of ISC 
+patterns. Contribute away ;)
+
+### Georg Grütter
+
+I propose we use a classification of the problem, a pattern tries to solve as
```

Comment:
I think a agree with this sentiment but would like to propose that "the problem" be defined as the intersection of vectors in any number of the planes that we're listing out below.